packages/devextreme/js/__internal/ui/popup/m_popup.ts

```diff
@@ -432,121 +467,244 @@ class Popup<
     this._observeContentResize(true);
   }
 
-  // @ts-expect-error ts-error
-  _renderContentImpl(): void {
-    this._renderTitle();
-    super._renderContentImpl();
+  _processContentRendering(): void {
+    this._renderTopToolbar();
+    this._renderBottomToolbar();
     this._renderResize();
-    this._renderBottom();
+    super._processContentRendering();
   }
 
-  _renderTitle(): void {
+  _getTopToolbarItems(): ToolbarItem[] {
+    const { showTitle, title } = this.option();
+    const { ios: isIOS } = devices.current();
+
     const items = this._getToolbarItems('top');
-    const { title, showTitle } = this.option();
 
-    if (showTitle && !!title) {
+    if (showTitle && Boolean(title)) {
       items.unshift({
-        location: devices.current().ios ? 'center' : 'before',
+        location: isIOS ? 'center' : 'before',
         text: title,
       });
     }
 
-    if (showTitle || items.length > 0) {
-      if (this._$title) {
-        this._$title.remove();
+    return items;
+  }
+
+  _renderTopToolbar(): void {
+    const { showTitle } = this.option();
+    const items = this._getTopToolbarItems();
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const shouldBeShown = showTitle || items.length > 0;
+
+    if (shouldBeShown) {
+      if (this._$topToolbar) {
+        this._updateToolbarOptions('top', { items });
+      } else {
+        this._renderTopToolbarImpl();
       }
-      const $title = $('<div>').addClass(POPUP_TITLE_CLASS).insertBefore(this.$content());
-      this._$title = this._renderTemplateByType('titleTemplate', items, $title).addClass(POPUP_TITLE_CLASS);
-      this._renderDrag();
-      this._executeTitleRenderAction(this._$title);
-      this._$title.toggleClass(POPUP_HAS_CLOSE_BUTTON_CLASS, this._hasCloseButton());
-    } else if (this._$title) {
-      this._$title.detach();
+      // To trigger toolbar width set in initial rendering to set menu button width (T1245421)
```

Comment:
Let's maybe move the commend to the method? It'll reduce copypaste and underline that it's related not to the call but to the method itself